Add release process documentation (#13)
ponylang/peg: RELEASE_PROCESS.md

# How to cut a peg release

> This document is aimed at members of the Pony team who might be cutting a release of Pony. It serves as a checklist that can take you through doing a release step-by-step.

Reviewer: Should this say "cutting a release of Pony"?

Author: Probably not. All the other libraries do right now. Lol.

Thoughts on how to change that in general?

Author: Is change to "peg" good enough?

Reviewer: Sure, either that or "a ponylang project" if you want to be more general

## Prerequisites

You must have commit access to the peg repository

## Releasing

> Please note that this document was written with the assumption that you are using a clone of the `peg` repo. You have to be using a clone rather than a fork. It is advised to your do this by making a fresh clone of the `peg` repo from which you will release.

Reviewer 2: ```suggestion
Please note that this document was written with the assumption that you are using a clone of the `peg` repo. You have to be using a clone rather than a fork. It is advised that you do this by making a fresh clone of the `peg` repo from which you will release.
```

Author: Welp, that mistake is in a ton of other documents.

Reviewer 2: I've missed it too.

```bash
git clone git@github.com:ponylang/peg.git peg-release-clean
cd peg-release-clean
```

Before getting started, you will need a number for the version that you will be releasing as well as an agreed upon "golden commit" that will form the basis of the release.

The "golden commit" must be `HEAD` on the `master` branch of this repository. At this time, releasing from any other location is not supported.

> For the duration of this document, that we are releasing version is `0.3.1`. Any place you see those values, please substitute your own version.

Reviewer 2: The phrasing is weird here. Here's a suggestion.

```suggestion
In this document, we pretend that the new release version is `0.3.1`. Wherever you see those values, please substitute with your own version.
```
